pkg/policy/visibility.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2019 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package policy
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/cilium/cilium/pkg/u8proto"
+)
+
+var annotationRegex = regexp.MustCompile(`^((<(Ingress|Egress)/([0-9]{1,6})/(TCP|UDP|ANY)/([A-Za-z]{3,32})>)(,(<(Ingress|Egress)/([0-9]{1,5})/(TCP|UDP|ANY)/([A-Za-z]{3,32})>))*)$`)
```

Comment:
nit: the port here is `[0-9]{1,6}`, but should it be `[1-9][0-9]{0,5}`? All-zeros port (or zero-prefixed port) is, I believe, nonsensical.

Also, not sure how the limits work here, but is this `6` inclusive or exclusive? I don't see a test with 6-digit input.

Any chance that the second half where it copies the same expression to allow a list of rediercts, could instead be represented using the `{1,X}` notation, where we'd figure out what the maximum is given the current character limits? I guess there's still the comma in between that makes it a bit trickier, hmm.. But I guess we could at least declare the regex for the annotation once, then combine strings here so we can ensure that they're the same (in case of any future changes). Example:

```
var (
    singleAnnotationRegex = '(<(Ingress|Egress)/([0-9]{1,6})/(TCP|UDP|ANY)/([A-Za-z]{3,32})>)'
    annotationRegex = regexp.MustCompile(fmt.Sprintf('^(%s)(,%s)*)$', singleAnnotationRegex, singleAnnotationRegex))
)
```